### PR TITLE
Changed slack antispam check to be less specific

### PR DESCRIPTION
**Pull request by starkmsu**

### CODE REVIEW

**Review by MikhailGerasimov — APPROVED** [review]

**Review by MikhailGerasimov — APPROVED** [review]

File: src/Lykke.Logs/SpamGuard.cs [hunk 1]
```diff
@@ -41,32 +41,43 @@ internal void SetMutePeriod(LogLevel level, TimeSpan mutePeriod)
             if (!_mutePeriods.ContainsKey(level))
                 return false;
 
+            var levelDict = _lastMessages.GetOrAdd(level, new Dictionary<string, DateTime>());
+            var key = GetEntryKey(component, process);
+            var now = DateTime.UtcNow;
+            DateTime lastTime = now;
+            lock(levelDict)
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Logs/SpamGuard.cs]: This and another lock of leveDict are both very short (no IO, only CPU inside lock), in this case SemaphorSlim could give significant performance gain. Consider this option, please

**Review by KonstantinRyazantsev — APPROVED** [review]

File: src/Lykke.Logs/SpamGuard.cs [hunk 2]
```diff
@@ -32,7 +33,7 @@ internal void SetMutePeriod(LogLevel level, TimeSpan mutePeriod)
             _mutePeriods[level] = mutePeriod;
         }
 
-        internal bool IsSameMessage(
+        internal bool ShouldBeMuted(
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Logs/SpamGuard.cs]: Could be made async

File: src/Lykke.Logs/SpamGuard.cs [hunk 3]
```diff
@@ -41,32 +42,52 @@ internal void SetMutePeriod(LogLevel level, TimeSpan mutePeriod)
             if (!_mutePeriods.ContainsKey(level))
                 return false;
 
+            var levelDict = _lastMessages.GetOrAdd(level, new Dictionary<string, DateTime>());
+            var key = GetEntryKey(component, process);
             var now = DateTime.UtcNow;
-            bool isSameMessage = false;
-            var messageInfo = new LastMessageInfo
+            DateTime lastTime;
+            _lock.Wait();
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Logs/SpamGuard.cs]: WaitAsync could be used

File: src/Lykke.Logs/SpamGuard.cs [hunk 4]
```diff
@@ -41,32 +42,52 @@ internal void SetMutePeriod(LogLevel level, TimeSpan mutePeriod)
...
+        }
+
+        public override Task Execute()
+        {
+            var now = DateTime.UtcNow;
+            foreach (var level in _lastMessages.Keys)
+            {
+                var levelDict = _lastMessages[level];
+                var mutePeriod = _mutePeriods[level];
+                _lock.Wait();
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Logs/SpamGuard.cs]: WaitAsync could be used

**Review by KonstantinRyazantsev — APPROVED** [review]

**Review by KonstantinRyazantsev — APPROVED** [review]

**Pull request merged**